testsuite/gnat2goto/tests/basic_exception/test.out

```diff
@@ -0,0 +1,49 @@
+Error from cbmc test:
+warning: ignoring typecast
```

Comment:
That's not good... why is this happening?

Reply:
I'm passing the `comment` as a `char[]` but CBMC is complaining that it's incompatible with `char*`. So I typecast it to `char*` and now it's complaining that the typecast is ignored.

Reply:
I couldn't reproduce this with C code alone. Maybe there is some processing when parsing C directly, that hides this cast.

Reply:
We used to have some issues with typecasts being not ignored as they should.  I think it might be worth digging a little deeper on this because errors like this used to mean the back-end was just ignoring that expression all together.